File: app/src/main/java/com/martinlaizg/geofind/data/access/database/converter/UserTypeTypeConverter.java
package com.martinlaizg.geofind.data.access.database.converter;

import androidx.room.TypeConverter;

import com.martinlaizg.geofind.data.enums.UserType;

public class UserTypeTypeConverter {

	@TypeConverter
	public static UserType toPlayLevel(String userTypeString) {
		return userTypeString == null ?
				null :
				UserType.valueOf(userTypeString);
	}

	@TypeConverter
	public static String fromPlayLevel(UserType userType) {
		return userType == null ?
				"" :
				userType.toString();
	}
}

File: app/src/main/java/com/martinlaizg/geofind/data/access/database/converter/UserTypeTypeConverter.kt
package com.martinlaizg.geofind.data.access.database.converter

import androidx.room.TypeConverter
import com.martinlaizg.geofind.data.enums.UserType

object UserTypeTypeConverter {
    @kotlin.jvm.JvmStatic
    @TypeConverter
    fun toPlayLevel(userTypeString: String?): UserType? {
        return if (userTypeString == null) null else UserType.valueOf(userTypeString)
    }

    @kotlin.jvm.JvmStatic
    @TypeConverter
    fun fromPlayLevel(userType: UserType?): String {
        return userType?.toString() ?: ""
    }
}